src/psyclone/psyir/transformations/read_only_verify_trans.py

```diff
@@ -0,0 +1,138 @@
+        return "ReadOnlyVerifyTrans"
+
+    def validate(self, node_list, options=None):
+        '''Performs validation checks specific to read-only-based
+        transformations.
+
+        :param node_list: the list of Node(s) we are checking.
+        :type node_list: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if distributed memory is configured.
+        :raises TransformationError: if transformation is applied to a \
+                                     Kernel or a BuiltIn call without its \
+                                     parent Loop.
+        :raises TransformationError: if transformation is applied to a Loop \
+                                     without its parent Directive when \
+                                     optimisations are applied.
+        :raises TransformationError: if transformation is applied to an \
+                                     orphaned Directive without its parent \
+                                     Directive.
+        '''
+
+        # Check ReadOnlyVerifyTrans specific constraints.
+
+        # Check constraints not covered by valid_node_types for
+        # individual Nodes in node_list.
+        from psyclone.psyir.nodes import Loop
+        from psyclone.psyGen import Directive, \
+            OMPParallelDirective, ACCParallelDirective
+
+        for node in node_list:
+
+            # Check that a ReadOnlyVerifyNode is not inserted between a Loop
+            # and its parent Directive when optimisations are applied, as
+            # this may result in including the end Directive for verification
+            # but not the beginning.
+            if isinstance(node, Loop) and isinstance(node.parent, Schedule) \
+               and isinstance(node.parent.parent, Directive):
+                raise TransformationError(
+                    "Error in {0}: Extraction of a Loop without its parent "
+                    "Directive is not allowed.".format(str(self.name)))
+
+            # Check that the ReadOnlyVerifyNode is not inserted within a thread
+            # parallel region when optimisations are applied. For instance,
+            # this may be between an orphaned Directive (e.g. OMPDoDirective,
+            # ACCLoopDirective) and its ancestor Directive (e.g. ACC or OMP
+            # Parallel Directive) or within an OMPParallelDoDirective.
+            if node.ancestor((OMPParallelDirective, ACCParallelDirective)):
+                raise TransformationError(
+                    "Error in {0}: Extraction of Nodes enclosed within "
+                    "a thread-parallel region is not allowed."
+                    .format(str(self.name)))
+
+        # Performs validation checks specific to PSyData-based
+        # transformations.
+        super(ReadOnlyVerifyTrans, self).validate(node_list, options)
```

Comment:
Please add an `__all__` at the end of this file.

Reply:
Done.